include/seqan3/std/ranges

```diff
@@ -173,8 +179,171 @@ template <typename type>
+
+/*!\interface std::ranges::end
+ * \brief Alias for ranges::end. Returns an iterator to the end of a range.
+ */
+using ::ranges::end;
+
+/*!\interface std::ranges::size
+ * \brief Alias for ranges::size. Obtains the size of a range whose size can be calculated in constant time.
+ */
+using ::ranges::size;
+
+/*!\interface std::ranges::copy
+ * \brief Alias for ranges::copy. Copies a range of elements to a new location.
+ */
+using ::ranges::copy;
+
+/*!\interface std::ranges::empty
+ * \brief Alias for ranges::empty. Checks whether a range is empty.
+ */
+using ::ranges::empty;
+
+/*!\interface std::ranges::equal
+* \brief Alias for ranges::equal. Determines if two sets of elements are the same.
+*/
+using ::ranges::equal;
+
+/*!\interface std::ranges::default_sentinel
+* \brief Alias for ranges::default_sentinel. Empty sentinel type for use with iterator types that know the bound of their range.
+*/
+using ::ranges::default_sentinel;
+
+/*!\interface std::ranges::sentinel_t
+* \brief Alias for ranges::sentinel_t. Obtains the sentinel type of a range.
+*/
+using ::ranges::sentinel_t;
+
+/*!\interface std::ranges::reference_t
+* \brief Alias for ranges::reference_t. Obtains a dereferenceable object's reference type.
+*/
+using ::ranges::reference_t;
+
+/*!\interface std::ranges::value_type_t
+* \brief Alias for ranges::value_type_t. Obtains the value type of a Readable type.
+*/
+using ::ranges::value_type_t;
+
+/*!\interface std::ranges::rvalue_reference_t
+* \brief Alias for ranges::rvalue_reference_t. Obtains a dereferenceable object's rvalue reference type.
+*/
+using ::ranges::rvalue_reference_t;
+
+/*!\interface std::ranges::difference_type_t
+* \brief Alias for ranges::difference_type_t. Obtains the difference type of a WeaklyIncrementable type.
+*/
+using ::ranges::difference_type_t;
+
+/*!\interface std::ranges::size_type_t
+* \brief Alias for ranges::size_type_t. Obtains the value type of a Size type.
+*/
+using ::ranges::size_type_t;
+
+/*!\interface std::ranges::iterator_t
+* \brief Alias for ranges::iterator_t. Obtains the iterator type of a range.
+*/
+using ::ranges::iterator_t;
 
+/*!\interface std::ranges::semiregular_t
+* \brief Alias for ranges::semiregular_t. Obtains the value type of a Semiregular type.
+*/
+using ::ranges::semiregular_t;
+
+/*!\interface std::ranges::common_reference_t
+* \brief Alias for ranges::common_reference_t. Determine the common reference type of a set of types.
+*/
+using ::ranges::common_reference_t;
+
+/*!\interface std::ranges::common_type_t
+* \brief Alias for ranges::common_type_t. Determine the common type of a set of types.
+*/
+using ::ranges::common_type_t;
+
+/*!\interface std::ranges::uncvref_t
+* \brief Alias for ranges::uncvref_t. Obtains the reference type of a top level cv.
+*/
+using ::ranges::uncvref_t;
+
+/*!\interface std::ranges::range_value_type_t
+* \brief Alias for ranges::range_value_type_t. Obtains the value type of a Range type.
+*/
+using ::ranges::range_value_type_t;
```

Comment:
use our `value_type_t` from core/metafunction instead

Reply:
But in core/metafunction, should it still reference std::ranges::value_type_t (in core/metafunction/iterator.hpp)

Reply:
No, it would be good to resolve to the actual functionality there. I will have a look if I can maybe get that done today.

Reply:
I have created #512